reference/7.1/Microsoft.PowerShell.Core/About/about_Tab_Expansion.md

```diff
@@ -0,0 +1,186 @@
+---
+ms.date: 07/22/2021
+description: Explains how to use the Tab Expansion feature in PowerShell.
+Locale: en-US
+online version: https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_tab_expansion?view=powershell-7.1&WT.mc_id=ps-gethelp
+schema: 2.0.0
+title: About tab expansion
+---
+# about_Tab_Expansion
+
+## Short description
+Command-line shells often provide a way to complete the names of long files or
+commands automatically, speeding up command entry and providing hints.
+PowerShell allows you to fill in filenames and cmdlet names by pressing the
+<kbd>Tab</kbd> key.
+
+## Long description
+
+Tab expansion is controlled by the internal function **TabExpansion** or
+**TabExpansion2**. Since this function can be modified or overridden, this
+discussion is a guide to the behavior of the default PowerShell configuration.
+
+> [!NOTE]
+> One limitation of the tab expansion process is that tabs are always
+> interpreted as attempts to complete a word. If you copy and paste command
+> examples into a PowerShell console, make sure that the sample does not
+> contain tabs; if it does, the results will be unpredictable and will almost
+> certainly not be what you intended.
+
+## File and cmdlet name completion
+
+To fill in a filename or path from the available choices automatically, type
+part of the name and press the <kbd>Tab</kbd> key. PowerShell will
+automatically expand the name to the first match that it finds. Pressing the
+<kbd>Tab</kbd> key repeatedly will cycle through all of the available choices.
+
+The tab expansion of cmdlet names is slightly different. To use tab expansion
+on a cmdlet name, type the entire first part of the name (the verb) and the
+hyphen that follows it. You can fill in more of the name for a partial match.
+For example, if you type `get-co` and then press the <kbd>Tab</kbd> key,
+PowerShell will automatically expand this to the `Get-Command` cmdlet (notice
+that it also changes the case of letters to their standard form). If you press
+<kbd>Tab</kbd> key again, PowerShell replaces this with the only other matching
+cmdlet name, `Get-Content`.
```

Comment:
This depends entirely on the PSReadLine setting. If you set `Set-PSReadLineKeyHandler -Key Tab -Function MenuComplete` you'll see an entirely different behaviour. But this is important to note because it's a very common preference (it's my default everywhere).

It's also worth noting that PSReadLine's completion behaviour is different between Windows and *nix. On Windows it will cycle through full completions, on *nix it will complete to the longest common prefix.

Reply:
Added note